File: resultcapture-service/src/main/java/com/devotion/capture/service/SurveyResultRepository.java
package com.devotion.capture.service;

import com.devotion.capture.model.SurveyResult;
import com.devotion.capture.model.QuestionAnswer;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface SurveyResultRepository extends MongoRepository<SurveyResult, String> {

    List<QuestionAnswer> findAnswersBySurveyId(String surveyId);

}

File: resultcapture-service/src/main/kotlin/com/devotion/capture/service/SurveyResultRepository.kt
package com.devotion.capture.service

import com.devotion.capture.model.QuestionAnswer
import com.devotion.capture.model.SurveyResult
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface SurveyResultRepository : MongoRepository<SurveyResult, String> {

    fun findAnswersBySurveyId(surveyId: String): List<QuestionAnswer>

}


